### ISSUE TITLE
several problems with ESLint newly installed with AirBnB through NPM

### ISSUE THREAD

**Issue by RobFosterNYC** [airbnb/javascript#2025]
1) - Why am I getting (Parse error) even though the code is correct.
.
![Screenshot_47](https://user-images.githubusercontent.com/18384135/55411621-f7c3ad00-5533-11e9-8eda-571132192992.png)

2) - .Also ever since I installed ESLint with AirBnB im getting other errors.
Such as errors for 2 spaces and im using 4.
How can I tell ESLint that I (WANT) to use the TAB key and keep my 4 spaces please.?

3) - Also,  im getting a not LF currently CRLF.
I NEVER had this problem before and now every one of my files of code has nothing but big red blocks on errors saying not LF currently CRLF..  (bangs head on keyboard).

4) - Also where and how do I change the Max Blank Lines.?

5) - Also, How can I use 'Single Quotes'  rather than "Double Quotes"
Because now ESLint is not recognizing that all of my code is using 'Single Quotes'.

6) - Also, how can I tell ESLint and AirBnB that I dont use Semi-Colons in my code.?
.

**Comment by ljharb** [comment]
That screenshot looks like your editor, not like the standard config file - rules usually aren’t configured in your editor, but per project.

Our config is set to 2 spaces; you’ll have to override the indent and jsx-indent rules if you want it set to 4 or to tabs.

All files, even on windows, should have LF line ending only. See #1224 for some workarounds.

For quotes, it the same as indent - you’ll have to override rules in your config file if you don’t want to use single quotes for JS and double quotes for jsx.

Similarly, our config requires semicolons. You’ll have to override those rules as well (but i strongly suggest you don’t override that one; using ASI comes with hazards that will only get worse as the language evolves).

**Comment by RobFosterNYC** [comment]
Thanks for responding @ljharb 
you forgot to mention where those files are located and what is the configuration line to achieve what I would like to do.  :)
I will read the #1224 link,  but I would love to have more info on the steps to do what I would like to do.
because I dont want to be editing the wrong file.
As far as the semi-colons I will hold off, but ive been speaking to many RL developers in my city and they are saying that most of doing away with semi-colons. - But I will hold off for the moment.

thanks.

**Comment by ljharb** [comment]
This stuff is all in the docs for eslint itself - make a `.eslintrc` file at the root of your project, with:
```sh
{
  "root": true,
  "extends": "airbnb", // or airbnb-base
  "rules": {
    // put your overrides here
  }
}
```
The LF rule is there to ensure that devs on Windows or non-Windows can collaborate easily; if everyone's using the same platform, problems don't tend to arise.

For semicolons, the vast majority definitely use semicolons (70% vs 30% on the last study of github code i've seen) but I wouldn't make your coding decisions solely or primarily based on popularity, in either direction.

I'm going to close this as answered, but please feel free to keep asking more questions here :-)

**Comment by RobFosterNYC** [comment]
So every time I make something new. even if its something simple.
I have to recreate a .eslintrc for each and every folder that is some kind of project ?
Is there a way I can just make it globally and "set it and forget it"  kinda thing ??
Instead of constantly having to recreate this .eslintrc over and over ??
Or am I am I wrong ??

**Comment by RobFosterNYC** [comment]
I tried many things, it doesnt work.
it wont respect my wishes on what I want it to do.  it keeps doing 2 spaces
and it also throws Parse Error's when there is nothing wrong with the code.
I regret installing this last night, its done nothing but caused me grief for the last 18 hours.

**Comment by ljharb** [comment]
Yes, exactly that. You don’t want it globally, because you want each project to independently and explicitly define its own configuration.

What’s your eslintrc look like? Try running eslint on the command line first - editor integrations are trickier to set up.

**Comment by RobFosterNYC** [comment]
I did, I installed it via Git Bash - npm install eslint -g
and then If I recall correctly I also did a  - eslint init
something to that affect,  its been an extremely long night, been at it for 18 hours
im tired, hungry and frustrated and very salty.
I was suppose to code last night, but spent it trying to fix this problem for 18 hours.  ggrrr.
.
![Screenshot_53](https://user-images.githubusercontent.com/18384135/55430762-e93dbb80-555c-11e9-8cf0-d5800a7bf43b.png)
